packages/protocol/src/routers/ConnextHandler.sol

```diff
@@ -131,16 +150,16 @@ contract ConnextHandler {
     external
     onlyConnextRouter
   {
-    if (!isTransferIdRecorded(transferId)) {
-      _failedTxns[transferId] =
-        FailedTxn(transferId, amount, asset, originSender, originDomain, actions, args);
-    }
+    uint256 nextNonce = getFailedTxnNextNonce(transferId);
+    _failedTxns[transferId][nextNonce] =
+      FailedTxn(transferId, amount, asset, originSender, originDomain, actions, args, nextNonce);
```

Comment:
@DaigaroCota can we add here an event with the same params? This will greatly facilitate the tracking of failed txns as we are only going to deal with this contract without the need to refer to "ConnextRouter".

Reply:
Requested event was added.